File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque
import sys

n, m, v = map(int, input().split())
graph = [[] for _ in range(n + 1)]
for _ in range(m):
    x, y = map(int, input().split())
    graph[x].append(y)
    graph[y].append(x)

for i in range(n + 1):
    graph[i].sort()

def dfs(n): 
    print(n, end = ' ')
    visited[n] =True
    
    for i in graph[n]:
        if not visited[i]:
            dfs(i)
    
def bfs(n):
    visited[n] = True
    q = deque([n])
    
    while q:
        tmp = q.popleft()
        print(tmp, end = ' ')
        for i in graph[tmp]:
            if not visited[i]:
                visited[i] = True
                q.append(i)
                
visited = [False for _ in range(n + 1)]
dfs(v)
print()
visited = [False for _ in range(n + 1)]
bfs(v)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.lang.reflect.Array;
import java.util.*;


public class Main {
    static int n, m, v = 0;
    static int[][] graph;
    static int[] visited;

    public void dfs(int v) {
        for (int i = 1; i <= n; i++) {
            if (graph[v][i] == 1 && visited[i] == 0) {
                System.out.print(i + " ");
                visited[i] = 1;
                dfs(i);
            }
        }
    }

    public void bfs(int v) {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(v);
        while (!queue.isEmpty()) {
            int tmp = queue.poll();
            for (int i = 1; i <= n; i++) {
                if (graph[tmp][i] == 1 && visited[i] == 0) {
                    visited[i] = 1;
                    queue.add(i);
                    System.out.print(i + " ");
                }
            }
        }
    }

    public static void main(String[] args) {
        Main t = new Main();
        Scanner scanner = new Scanner(System.in);

        n  = scanner.nextInt();
        m = scanner.nextInt();
        v = scanner.nextInt();
        graph = new int[n + 1][n + 1];
        for (int i = 0; i < m; i++) {
            int a = scanner.nextInt();
            int b = scanner.nextInt();
            graph[a][b] = 1;
            graph[b][a] = 1;
        }

        visited = new int[n + 1];
        System.out.print(v + " ");
        visited[v] = 1;
        t.dfs(v);
        System.out.println();

        visited = new int[n + 1];
        System.out.print(v + " ");
        visited[v] = 1;
        t.bfs(v);
    }

}